### PR TITLE
fix case where TTML data is an empty string

### PR DESCRIPTION
**Pull request by fadomire**
when using DOMParser parseFromString with an empty string it returns an errored XML document
the next part of the code is then executed and throw an unnecessary error

### CODE REVIEW

File: lib/text/ttml_text_parser.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ shaka.text.TtmlTextParser = class {
     const parser = new DOMParser();
     let xml = null;
 
+    // dont try to parse empty string as 
+    // DOMParser will not throw error but return an errored xml
+    if (str == '') return ret
```
Comment (TheModMaker) [inline lib/text/ttml_text_parser.js]: This needs curly brackets and needs to be on the next line.

**Comment by fadomire** [comment]
ok will do

**Comment by fadomire** [comment]
@TheModMaker hello, i finally took the time to write the unit test and fix the code style
Let me know if you want me to squash commits or do other changes

Thanks

**Comment by TheModMaker** [comment]
It looks like there is an explicit test to reject the empty string.  You'll need to edit that.  The other failure should be fixed with a rebase.

**Comment by fadomire** [comment]
ok thanks, i should have seen that another test was checking empty string and expected a rebase would be needed :)
thanks for your time, i’ll fix that next week

**Comment by fadomire** [comment]
@TheModMaker i rebased and removed part of unit test that was conflicting
